docs/encryption.rst

```diff
@@ -0,0 +1,252 @@
+Client-Side Field-Level Encryption (CSFLE) and Queryable Encryption (QE)
+============================================================
+
+This page documents how to configure and use MongoDB Client-Side Field-Level Encryption (CSFLE) and Queryable Encryption (QE) in DoctrineMongoDBBundle.
+
+.. note::
+
+    CSFLE and QE are advanced MongoDB features that allow you to encrypt specific fields in your documents, with optional support for searching encrypted data (Queryable Encryption).
+
+Configuration
+-------------
+
+.. tip::
+
+    For a general overview of configuration options, see :doc:`config`.
+
+To enable CSFLE or QE, you need to configure the ``autoEncryption`` driver option under your connection's ``driver_options``. At a minimum, you must specify the ``keyVaultNamespace`` and ``kmsProviders``. Additional options are available for advanced use cases.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        doctrine_mongodb:
+            connections:
+                default:
+                    server: "mongodb://localhost:27017"
+                    driver_options:
+                        autoEncryption:
+                            keyVaultNamespace: "encryption.__keyVault"
+                            kmsProviders:
+                                local:
+                                    key: "YOUR_BASE64_KEY"
+                            # Optional: see below for more options
+
+    .. code-block:: php
+
+        use Symfony\Config\DoctrineMongodbConfig;
+
+        return static function (DoctrineMongodbConfig $config): void {
+            $config->connection('default')
+                ->server('mongodb://localhost:27017')
+                ->driverOptions([
+                    'autoEncryption' => [
+                        'keyVaultNamespace' => 'encryption.__keyVault',
+                        'kmsProviders' => [
+                            'local' => [
+                                'key' => 'YOUR_BASE64_KEY',
+                            ],
+                        ],
+                        // ... other options ...
+                    ],
+                ]);
+        };
+
+Supported KMS Providers
```

Comment:
Do we want to duplicate the full list here? If so, we may want to properly explain the individual options required for each KMS provider and where to obtain them.